tests/foreman/cli/test_auth.py

```diff
@@ -0,0 +1,415 @@
+               on next command run
+            3. Wait until session expires
+
+        :expectedresults: Session expires after specified time
+            and saved credentials are not applied
+
+        """
+        try:
+            idle = Settings.list({'search': 'name=idle_timeout'})[0][u'value']
+            Settings.set({'name': 'idle_timeout', 'value': 1})
+            result = configure_sessions(add_default_creds=True)
+            self.assertEqual(result, 0, 'Failed to configure hammer sessions')
+            Auth.login({
+                'username': self.uname_admin,
+                'password': self.password
+            })
+            result = Auth.with_user().status()
+            self.assertIn(
+                LOGEDIN_MSG.format(self.uname_admin),
+                result[0][u'message']
+            )
+            # list organizations without supplying credentials
+            with self.assertNotRaises(CLIReturnCodeError):
+                Org.with_user().list()
+            # wait until session expires
+            sleep(70)
+            with self.assertRaises(CLIReturnCodeError):
+                Org.with_user().list()
+            result = Auth.with_user().status()
+            self.assertIn(
+                LOGEDOFF_MSG.format(self.uname_admin),
+                result[0][u'message']
+            )
+        finally:
+            # reset timeout to default
+            Settings.set({'name': 'idle_timeout', 'value': '{}'.format(idle)})
+
+    @tier1
+    def test_negative_no_credentials(self):
+        """Attempt to execute command without authentication
+
+        :id: 8a3b5c68-1027-450f-997c-c5630218f49f
+
+        :expectedresults: Command is not executed
+        """
+        result = configure_sessions(False)
+        self.assertEqual(result, 0, 'Failed to configure hammer sessions')
+        result = Auth.with_user().status()
+        self.assertIn(
+            NOTCONF_MSG.format(self.uname_admin),
+            result[0][u'message']
+        )
+        with self.assertRaises(CLIReturnCodeError):
+            Org.with_user().list()
+
+    @tier1
+    def test_negative_no_permissions(self):
+        """Attempt to execute command out of user's permissions
+
+        :id: 756f666f-270a-4b02-b587-a2ab09b7d46c
+
+        :expectedresults: Command is not executed
+
+        """
+        result = configure_sessions()
+        self.assertEqual(result, 0, 'Failed to configure hammer sessions')
+        Auth.login({'username': self.uname_viewer, 'password': self.password})
+        result = Auth.with_user().status()
+        self.assertIn(
+            LOGEDIN_MSG.format(self.uname_viewer),
+            result[0][u'message']
+        )
+        # try to update user from viewer's session
+        result = User.with_user().update({
+            'login': self.uname_admin,
+            'new-login': gen_string('alpha'),
+        }, return_raw_response=True)
+        self.assertNotEqual(result.return_code, 0)
```

Comment:
why we are using raw response we could use simple response and self.assertRaises(CLIReturnCodeError)

Reply:
And no point in extending base `update()` with `return_raw_response` arg then.